src/olm/account/mod.rs

```diff
@@ -682,16 +792,163 @@ mod libolm {
     }
 }
 
+mod dehydrated_device {
+    use matrix_pickle::{Decode, DecodeError, Encode, EncodeError};
+    use zeroize::{Zeroize, ZeroizeOnDrop};
+
+    use super::{
+        fallback_keys::{FallbackKey, FallbackKeys},
+        one_time_keys::OneTimeKeys,
+        Account,
+    };
+    use crate::{
+        types::{Curve25519Keypair, Curve25519SecretKey},
+        DehydratedDeviceError, Ed25519Keypair, KeyId,
+    };
+
+    #[derive(Encode, Decode, Zeroize, ZeroizeOnDrop)]
+    pub(crate) struct OneTimeKey {
+        pub(crate) private_key: Box<[u8; 32]>,
+    }
+
+    impl From<&OneTimeKey> for FallbackKey {
+        fn from(key: &OneTimeKey) -> Self {
+            FallbackKey {
+                key_id: KeyId(0),
+                key: Curve25519SecretKey::from_slice(&key.private_key),
+                published: true,
+            }
+        }
+    }
+
+    impl TryFrom<&FallbackKey> for OneTimeKey {
+        type Error = ();
+
+        fn try_from(key: &FallbackKey) -> Result<Self, ()> {
+            Ok(OneTimeKey { private_key: key.secret_key().to_bytes() })
+        }
+    }
+
+    #[derive(Zeroize, ZeroizeOnDrop)]
+    pub(crate) struct OptFallbackKey {
+        pub(crate) fallback_key: Option<OneTimeKey>,
```

Comment:
Not something for now, but perhaps it'll make sense to implement `Decode` and `Encode` support for `Option<T>` in `matrix-pickle`.

Reply:
Yup, I was considering that, but figured it would be easier to do it here for now.  Maybe the next time we need to support an `Option<T>`.